ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -963,6 +963,29 @@ - (long long) getFirstInstallTime {
     return [@(floor([installDate timeIntervalSince1970] * 1000)) longLongValue];
 }
 
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(getStartupTimeSync) {
+    return @(self.getStartupTime);
+}
+
+RCT_EXPORT_METHOD(getStartupTime:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+    resolve(@(self.getStartupTime));
+}
+ 
+- (long long) getStartupTime {
+    size_t len = 4;
+    int mib[len];
+    struct kinfo_proc kp;
+
+    sysctlnametomib("kern.proc.pid", mib, &len);
+    mib[3] = getpid();
+    len = sizeof(kp);
+    sysctl(mib, 4, &kp, &len, NULL, 0);
+
+    struct timeval startTime = kp.kp_proc.p_un.__p_starttime;
+    double startTimeMilliSeconds = startTime.tv_sec * 1e3 + startTime.tv_usec / 1e3;
+    return [@(floor(startTimeMilliSeconds)) longLongValue];
```

Comment:
wouldn't the following be simpler?

```obj-c
NSDate *processStartDate = [[NSProcessInfo processInfo] systemUptime];
NSTimeInterval processStartTime = [[NSDate date] timeIntervalSince1970] - [processStartDate timeIntervalSince1970];
```

Reply:
Agreed - @RikSchefferAmsterdam if you altered this to let the platform handle the mathematical gymnastics, this would be a lot easier to approve + merge

Alternatively, if you had a very good reason to implement it as you did, then leaving a comment specifically mentioning why you did not use NSProcessInfo and timIntervalSince1970 methods would be useful

Reply:
Sorry it took so much time to investigate this from my side.

My implementation retreives the starttime of the app process, which is more accurate than comparing the current time with the system startuptime. Specifically when you want to track the app launch time, you would want the starttime of the process. 

I have not found a way to retreive the app process startup time with NSProcessInfo.

I will add some comments in the code to explain this.

Reply:
that is a reasonable defense of the style :-), works for me, thank you